src/kernel/notifications_interface.h

```diff
@@ -32,6 +33,12 @@ class Notifications
     //! occurred while flushing block data to disk. Functions notifying about
     //! this error may not return an error code or raise an exception.
     virtual void flushError(const std::string& debug_message) {}
+
+    //! The fatal error notification is sent to notify the user and start
+    //! shutting down if an error happens in kernel code that can't be recovered
+    //! from. After this notification is sent, whatever function triggered the
+    //! error should also return an error code or raise an exception.
```

Comment:
In commit "kernel: Add fatalError method to notifications" (e0653c2223f8d6d74645c9b5742de1c069d8c3a7)

Maybe add "This notification can be used to log the error, or notify the user, trigger an application shutdown to prevent causing more errors."